File: MusicStore.Microservices.Products.Data/Models/Product.cs
using MusicStore.Microservices.Products.Data.Models.Base;

namespace MusicStore.Microservices.Products.Data.Models;

public class Product : IEntity
{
    public string Name { get; set; }

    public string Description { get; set; }

    public double Price { get; set; }

    public string ImageUrl { get; set; }

    public long Stock { get; set; } = 100;
    public Guid Id { get; set; }
}

File: Frontend/src/app/models/product.ts
export class Product {
    id: string;
    name: string;
    description: string;
    price: number;
    imageUrl: string;
    stock: number;

    constructor(id: string, name: string, description: string, price: number, imageUrl: string, stock: number) {
        this.id = id;
        this.name = name;
        this.description = description;
        this.price = price;
        this.imageUrl = imageUrl;
        this.stock = stock;
    }
}
